backends-clickhouse/src/test/scala/io/glutenproject/execution/GlutenClickHouseTPCHParquetAQESuite.scala

```diff
@@ -330,5 +337,44 @@ class GlutenClickHouseTPCHParquetAQESuite
       }
     )
   }
+
+  test("issue-3049 local properties level controller statement fallback") {
+    val sqlFile = tpchQueries + "/q01.sql"
+    val sqlStr = Source.fromFile(new File(sqlFile), "UTF-8").mkString
+    var expected: Seq[Row] = null
+    withSQLConf(vanillaSparkConfs(): _*) {
+      val df = spark.sql(sqlStr)
+      expected = df.collect()
+    }
+
+    val fthread = new Thread(
```

Comment:
As the proposed control is bound to a thread instead of a query, it may be nice to add some more test in this thread, i.e., explicitly enable gluten, then execute a query and validate it.